include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -502,14 +503,17 @@ namespace internal
 
       std::map<unsigned int, std::vector<char>> send_data;
 
-      for (const auto &i : send_data_temp)
+      for (const typename std::map<unsigned int,
+                                   std::vector<ConstraintLine>>::value_type &i :
+           send_data_temp)
```

Comment:
I think one could argue about this case and a few similar ones below, as the variable `send_data_temp` is given the full type at the construction in line 480, but I'm not going to argue against it since I typically am in favor of spelling out types.